### ISSUE TITLE
Auto Linking rn-secure-storage not working for iOS

### ISSUE THREAD

**Issue by cakesoft-utkarsh** [talut/rn-secure-storage#73]
Adding the package in a react-native 0.71.13 app does not link the package on iOS.

This behaviour is reproducible on a fresh empty react-native application.

**Comment by Martinocom** [comment]
There are also problems on RN 0.72.4. I have a Mac with M2.
I tried with both `2.0.7` and `2.0.8` versions.

Android works without problems
```
    buildToolsVersion = "33.0.0"
    minSdkVersion = 21
    compileSdkVersion = 33
    targetSdkVersion = 33
    kotlinVersion = "1.7.20"
    ndkVersion = "23.1.7779620"
```
iOS has problems.
```
    target = 13.1
    platform = iPhone
    ruby = (rbenv) 2.7.8
```

Our iOS app login correctly, the obtained token is valid but when calling RNSecureStorage, both `.get` or `.set`methods makes on JS Thread an exception: `[TypeError: Cannot read property 'get' of null]`

When printing just `console.log("Trying to find the token", RNSecureStorage)`, the results from both platforms are:

```
// Android
Trying to find the token {"exists": [Function promiseMethodWrapper], "get": [Function promiseMethodWrapper], "getConstants": [Function anonymous], "remove": [Function promiseMethodWrapper], "set": [Function promiseMethodWrapper]}

// iOS
Trying to find the token null
```

Linking manually the library does not solve the problem.

**Comment by talut** [comment]
I just released v3.0.0 I hope it'll solve your problems.

Please check the new version and Readme file. It also includes a few new features as well.

It's tested with newest RN versions.
